app/controllers/races_controller.rb

```diff
@@ -0,0 +1,29 @@
+class RacesController < ApplicationController
+  before_action :set_race, only: [:show]
+  before_action :set_race_gon, only: [:show]
+
+  def show
+    render template: 'races/show'
```

Comment:
Is this line necessary? `races/show` should be the default render for `RacesController#show`

Reply:
Artifact of when it was 3 different controllers.  Based on the other discussion we had, are we going to make this 3 different actions, one for `race`, `bingo`, and `randomizer` respectively?

Reply:
Yeah I do like the idea of passing the race type into the controller in a more structured way than a string in the params hash. I was thinking three different actions initially, like `race_show`, `bingo_show`, `randomizer_show`, but actually what would you think of a setup like this?

```ruby
# config/routes.rb
get '/races/:race',       to: 'races/races#show',       as: :race
get '/bingos/:race',      to: 'races/bingos#show',      as: :bingo
get '/randomizers/:race', to: 'races/randomizers#show', as: :randomizer

# app/controllers/races/application_controller.rb
class Races::ApplicationController < ApplicationController
  before_action :set_race,     only: [:show]
  before_action :set_race_gon, only: [:show]

  def show
  end

  private

  def set_race_gon
    # ...
  end
end

# app/controllers/races/races_controller.rb
class Races::RacesController < Races::ApplicationController
  def show
    render 'races/show'
  end

  private

  def set_race
    @race = Race.find(params[:id])
  end
end

# app/controllers/races/bingos_controller.rb
class Races::BingosController < Races::ApplicationController
  def show
    render 'races/show'
  end

  private

  def set_race
    @race = Bingo.find(params[:id])
  end
end

# app/controllers/races/randomizers_controller.rb
class Races::RandomizersController < Races::ApplicationController
  def show
    render 'races/show'
  end

  private

  def set_race
    @race = Randomizer.find(params[:id])
  end
end
```
This subfolder-based ApplicationController setup is how we do APIs and our nested non-API runs controllers (`runs/exports`, `runs/likes`).

I don't know if I caught what your three controllers were like before they were merged; did you initially try something like the above then find something bad about it? Or was it more copypasta than this?

Reply:
It was almost identical but wasn't namespaced which seemed like it was just cluttering everything up and didn't inherit from a shared parent like that.  I think those 2 things would solve the style issues I was having with it while being a lot more consistent with how you would expect things to work.